HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs

```diff
@@ -78,6 +85,8 @@ public ActionResult AdvanceSearchExaminations(ExaminationSearchDTO examinationSe
         /// </summary>
         /// <param name="id">id of the object to be changed</param>
         /// <returns>if alright returns code 200(Ok), if not 400(bed request), if connection lost returns 500</returns>
+        /// 
+        [Authorize(Roles = UserRoles.Patient)]
         [HttpPut("cancel/{id}")]
         public ActionResult CancelExamination(int id)
         {
```

Comment:
You also need a jmbg check here, patients can only cancel their own examinations.